barman/retention_policies.py

```diff
@@ -350,6 +391,16 @@ def _backup_report(self, source):
                         found = True
             else:
                 report[bid] = BackupInfo.NONE
+
+            if backups[bid].has_children:
+                status = report[bid]
+                if report[bid] in (BackupInfo.KEEP_STANDALONE, BackupInfo.KEEP_FULL):
```

Comment:
Also here

Reply:
Right, makes sense at a first glance.

Reply:
```suggestion
                if status in (BackupInfo.KEEP_STANDALONE, BackupInfo.KEEP_FULL):
```

Same comment about the lookup.